contracts/sdk/indexer/backfill.ts

```diff
@@ -25,42 +23,17 @@ async function main() {
     });
     let db = await bootstrap();
     console.log("Finished bootstrapping DB");
-    Gummyroll = loadProgram(
-        provider,
-        GUMMYROLL_PROGRAM_ID,
-        "target/idl/gummyroll.json"
-    ) as anchor.Program<Gummyroll>;
-    Bubblegum = loadProgram(
-        provider,
-        BUBBLEGUM_PROGRAM_ID,
-        "target/idl/bubblegum.json"
-    ) as anchor.Program<Bubblegum>;
+
+    const parserState = loadPrograms(provider);
     console.log("loaded programs...");
-    console.log("Filling in gaps for tree:", treeId);
 
-    // Get first gap
-    const trees = await db.getTrees();
-    const treeInfo = trees.filter((tree) => (tree[0] === treeId));
-    let startSeq = 0;
-    let startSlot: number | null = null;
-    if (treeInfo) {
-        let [missingData, maxDbSeq, maxDbSlot] = await db.getMissingData(
-            0,
-            treeId
-        );
-        console.log(missingData, maxDbSeq, maxDbSlot);
-        if (missingData.length) {
-            startSlot = missingData[0].prevSlot;
-            startSeq = missingData[0].prevSeq;
-        } else {
-            startSlot = maxDbSlot;
-            startSeq = maxDbSeq;
-        }
-    }
+    // Fill gaps
+    console.log("Filling in gaps for tree:", treeId);
+    let { maxSeq, maxSeqSlot } = await fillGapsTx(connection, db, parserState, treeId);
 
-    // Backfill
-    console.log(`Starting from slot!: ${startSlot} `);
-    const maxSeq = await backfillTreeHistory(connection, db, { Gummyroll, Bubblegum }, treeId, startSeq, startSlot);
+    // Backfill to on-chain state, now with a complete db
```

Comment:
Does any of this newer logic need to be added to the process in indexer.ts?

Reply:
It is automatically reflected in indexer because the parsing changes are implemented at the slot-fetching level